Documentation/contributing/docs/docsframework.rst

```diff
@@ -0,0 +1,382 @@
+------------------------
+
+We do not implement custom directives as of this writing.
+
+Custom extensions
+-----------------
+
+We use custom extensions for Sphinx, implemented under ``Documentation/_exts``.
+
+- One defines the custom filters for the spell checker.
+- One patches Sphinx's HTML translator to open all external links in new tabs.
+
+Google Analytics
+----------------
+
+The documentation uses Google Analytics to collect metrics. This is configured
+in ``Documentation/conf.py``.
+
+Customization
+-------------
+
+Additional elements of customization for Cilium's documentation and defined in
+the main repository include:
+
+- Some custom CSS; see also class ``wrapped-table`` in the related CSS file
+  ``Documentation/_static/wrapped-table.css``
+
+- A "Copy" button, including a button to copy only commands from console-code
+  blocks, implemented in ``Documentation/_static/copybutton.js`` and
+  ``Documentation/_static/copybutton.css``
+
+- Custom header and footer definitions, for example to make link to Slack
+  target available on all pages
+
+- Warning banner on older branches, telling to check out the latest version
+  (these may be handled directly in the ReadTheDocs configuration in the
+  future, see also :gh-issue:`29969`)
+
+:spelling:ignore:`Algolia` search engine
+----------------------------------------
+
+- *To be documented.*
+
+Build set up
+============
+
+.. _docs_requirements:
+
+Requirements (dependencies)
+---------------------------
+
+We have two files for requirements: one where we declare and pin the core
+dependencies for the documentation build system, and that we use to generate a
+second requirement files that includes all sub-dependencies:
+
+- The base requirements are defined in
+  ``Documentation/requirements-min/requirements.txt``.
+- By running ``make -C Documentation update-requirements``, we use this file to
+  generate ``Documentation/requirements.txt``.
+
+Dependencies defined in ``Documentation/requirements-min/requirements.txt``
+should never be updated in ``Documentation/requirements.txt`` directly.
+Instead, update the former and regenerate the latter.
+
+These requirements are used to build the ``docs-builder`` Docker image.
+
+One of the dependency is the custom theme that we use for the documentation.
+
+Docker set-up
+-------------
+
+The documentation build system relies on a Docker image, ``docs-builder``, to
+ensure that the build environment is consistent across different systems.
+Resources related to this image include ``Documentation/Dockerfile`` and the
+requirement files.
+
+Versions of this image are automatically built and published to a registry when
+the Dockerfile or the list of dependencies is updated. This is handled in CI
+workflow ``.github/workflows/build-images-docs-builder.yaml``.
+
+The deployment described in this workflow should be run twice on each PR
+updating the Dockerfile or its dependencies, to ensure that the CI picks up an
+updated image.
```

Comment:
Let's be clearer to users about what to do
```suggestion
If a PR updates the Dockerfile or its dependencies, run the deployment described
in this workflow twice to ensure that the CI picks up an updated image.
```

Reply:
OK but not all users can do that, so it will be along “run or have someone run”